modules/nextflow/src/main/groovy/nextflow/script/ProcessConfig.groovy

```diff
@@ -931,4 +932,25 @@ class ProcessConfig implements Map<String,Object>, Cloneable {
         return this
     }
 
+    ProcessConfig arch( Map params, value )  {
+        if( value instanceof String ) {
+            if( params.name==null )
+                params.name=value
+        }
+        else if( value != null )
+            throw new IllegalArgumentException("Not a valid `arch` directive value: $value [${value.getClass().getName()}]")
+        arch(params)
+        return this
+    }
+
+    ProcessConfig arch( value ) {
+        if( value instanceof String )
+            configProperties.put('arch', [name: value])
+        else if( value instanceof Map )
+            configProperties.put('arch', value)
+        else if( value != null )
+            throw new IllegalArgumentException("Not a valid `arch` directive value: $value [${value.getClass().getName()}]")
+        return this
+    }
```

Comment:
Worth adding a test in `ProcessConfigTest`

Reply:
done!